### PR TITLE
[top, ast] Generate open source ast registers in topgen

### PR DESCRIPTION
**Pull request by tjaychen**
- generate open source ast registers as part of topgen
- remove the need for xbar_peri to hardcode ast address
- introduce different module attributes to distinguish
  between normal modules, templated modules, and top specific
  modules like sensor_ctrl and ast

### CODE REVIEW

**Comment by tjaychen** [comment]
My names for module attributes are...horrendous... if anyone has suggestions for better names.. do let me know.

**Review by msfschaffner — APPROVED** [review]
Thanks, I like the consolidation of these special keys into one attribute key.
Just a comment about naming.

File: hw/top_englishbreakfast/data/top_englishbreakfast.hjson [hunk 1]
```diff
@@ -198,9 +198,13 @@
 
   // `module` defines the peripherals.
   // Details are coming from each modules' config file `ip.hjson`
-  // TODO: Define parameter here
-  // generated: A module is templated and generated as part of topgen
-  // top_only: A module is not templated but is specific to 'top_*' instead of 'ip'
+  // attr: There are a few types of modules supported
+     //   normal(default): Normal, non-templated modules that will be instantiated
+     //   templated: These modules are templated and must be run through topgen
+     //   non_templ: These modules are not templated, but need to have reggen run
```
[the remark this answers is not in the document]
Comment (tjaychen) [reply hw/top_englishbreakfast/data/top_englishbreakfast.hjson:204]: yeah they were intentional...although i'm not sure why now.. haha

yeah i thought about `reggen_top`, but i wanted to emphasize its...non-template-ness somehow.
Although if you think that's clearer, i can just go with it.

**Comment by msfschaffner** [comment]
LGTM, thanks

**Pull request merged**